### ISSUE TITLE
Stylo: Implement column-fill

### ISSUE THREAD

**Issue by Manishearth** [servo/servo#13764]
Please go through [the guide](https://github.com/servo/servo/wiki/Stylo-hacking-guide) if you wish to hack on stylo.

[`column-fill`](https://developer.mozilla.org/en-US/docs/Web/CSS/column-fill) is a keyword property ([implementation guide](https://github.com/servo/servo/wiki/Stylo-hacking-guide#implementing-a-simple-keyword-property)) with the set of possible values being `auto | balance`. `auto` is the default.
The constant prefix is [`NS_STYLE_COLUMN_FILL`](https://dxr.mozilla.org/mozilla-central/rev/7452437b3ab571b1d60aed4e973d82a1471f72b2/layout/style/nsStyleConsts.h#1066), but you should not need to explicitly specify it.

File: https://github.com/servo/servo/blob/master/components/style/properties/longhand/column.mako.rs

Testcase:

```html
<style>

.example {
  -moz-column-count: 4;
  column-count: 4;
  columns: 4;
  -moz-columns: 4;
  height: 105px;
  border: 1px solid rgba(0, 0, 0, 0.1);
  padding: 0 0.5em;
}

.example-balance {
  -webkit-column-fill: balance;
  -moz-column-fill: balance;
  column-fill: balance;
  margin-bottom: 1em;
}

.example-auto {
  -webkit-column-fill: auto;
  -moz-column-fill: auto;
  column-fill: auto;
}

body {
  font-size: 12px;
  font-family: 'Georgia', serif;
  font-weight: 400;
  line-height: 1.45;
  color: #333;
  background: #ecf0f1;
  padding: 1em;
}

code {
  font-family: "Courier New", monospace;
}

</style>
<code>column-fill: balance;</code>
<div class="example example-balance">
  <p> Lorem ipsum dolor sit amet, consectetur adipiscing elit. Phasellus libero libero, placerat at hendrerit nec, vulputate consectetur nisi. Morbi scelerisque lectus id sapien laoreet accumsan. Nunc eget tincidunt ligula, eget suscipit tellus. Sed iaculis nibh gravida faucibus porta. Sed lacinia tristique elementum. Etiam odio sem, dapibus eu tempus vel, consequat non turpis.</p>

</div>
<code>column-fill: auto;</code>
<div class="example example-auto">
  <p> Lorem ipsum dolor sit amet, consectetur adipiscing elit. Phasellus libero libero, placerat at hendrerit nec, vulputate consectetur nisi. Morbi scelerisque lectus id sapien laoreet accumsan. Nunc eget tincidunt ligula, eget suscipit tellus. Sed iaculis nibh gravida faucibus porta. Sed lacinia tristique elementum. Etiam odio sem, dapibus eu tempus vel, consequat non turpis.</p>
</div>
```

The testcase should look the same in Firefox and Stylo, with the columns having roughly the same amount of text in the first block and an imbalance in the second.

**Comment by willempienaar** [comment]
I'd like to work on this issue please.

**Comment by Manishearth** [comment]
Cool! Let us know if you need help.

**Comment by Wafflespeanut** [comment]
@willempienaar Any news on this?

**Comment by willempienaar** [comment]
@Wafflespeanut No feedback yet. I have next week off so I am planning to finish it then.

**Comment by willempienaar** [comment]
Haven't gotten around to this one. If somebody wants to grab it, then it is available. Otherwise I will continue next week.

**Comment by KiChjang** [comment]
Let us know if you're working on this again.

**Comment by Haggus** [comment]
Do you mind if I hack on this?

**Comment by KiChjang** [comment]
@Haggus Go for it!

**Comment by Haggus** [comment]
I went through style guide and added column-fill. Tested and compared it to Firefox. It's not "exactly" the same, but very similar. Let me know if there is more changes I need to do. Also, should the test also be added into the code?

**Comment by Manishearth** [comment]
Do you have screenshots?

**Comment by Haggus** [comment]
Firefox:
![column-fill-firefox](https://cloud.githubusercontent.com/assets/8225391/19832406/40a8e4f6-9e23-11e6-9e8e-b448ea8c78db.png)

Stylo (after my changes):
![column-fill-stylo](https://cloud.githubusercontent.com/assets/8225391/19832405/40a8912c-9e23-11e6-810f-d639c6bc11a2.png)

**Comment by Manishearth** [comment]
Yep, this is correct. The difference happens because of that extra line before the lorem ipsum in firefox (might be due to some other unimplemented property). That's ok.

**Comment by KiChjang** [comment]
Fixed in #13978.
